Source/common/SNTConfigurator.m

```diff
@@ -322,70 +354,85 @@ - (NSString *)machineID {
 }
 
 - (void)reloadConfigData {
-  NSFileManager *fm = [NSFileManager defaultManager];
-  if (![fm fileExistsAtPath:self.configFilePath]) {
-    // As soon as saveConfigToDisk is called, reloadConfigData will be called again because
-    // of the SNTFileWatchers on the config path. No need to use dictionaryWithCapacity: here.
-    self.configData = [NSMutableDictionary dictionary];
+  // Load the mobileconfig
+  self.configData = [self mobileConfig];
+  if (!self.configData[kClientModeKey]) {
+    // Default to Monitor if the config is missing or invalid
     self.configData[kClientModeKey] = @(SNTClientModeMonitor);
-    [self saveConfigToDisk];
-    return;
-  };
+  }
+
+  // Nothing else to do if a sync server is not involed
+  if (!self.configData[kSyncBaseURLKey]) return;
+
+  // Load the last known sync state
+  NSMutableDictionary *syncState =
```

Comment:
It looks like the syncState variable is always reassigned in line 393 before it is ever used, so you could just get rid of it and change line 397 to read directly from syncStateData.

Reply:
YES